cmd/deployment-check/input.go

```diff
@@ -182,8 +182,9 @@ func parseInputValues() {
 	}
 	if deadline > 0 {
 		log.Infoln("Parsed check deadline time from the environment:", deadline)
-		checkTimeLimit = time.Duration((int64(deadline) - time.Now().Unix()) * 1e9) // Multiply by 1,000,000,000 because that's how many nanoseconds are in a second
+		checkTimeLimit = time.Duration((int64(deadline) - time.Now().Unix() - 5) * 1e9) // Multiply by 1,000,000,000 because that's how many nanoseconds are in a second
```

Comment:
Can you refactor this to parse the deadline into a `time.Time`, then use `time.After()` to determine if the time has passed?

You can also use `time.Sub()` to remove a duration (like `time.Second * 5`) from a `time.Time`.